lib/livebook_web/live/session_helpers.ex

```diff
@@ -21,13 +21,25 @@ defmodule LivebookWeb.SessionHelpers do
 
     case Livebook.Sessions.create_session(opts) do
       {:ok, session} ->
-        push_redirect(socket, to: Routes.session_path(socket, :page, session.id))
+        redirect_path =
+          socket
+          |> Routes.session_path(:page, session.id)
+          |> maybe_add_anchor_link(opts)
```

Comment:
We can share some logic by defining a helper `session_path(socket, session_id, opts)` and passing `url_hash:` as an option from both places.